api/v1alpha1/k8s_types.go

```diff
@@ -103,6 +103,19 @@ type PersistentVolumeClaimSpec struct {
 	VolumeMode *PersistentVolumeMode `json:"volumeMode,omitempty"`
 }
 
+// VolumeSpec contains the Persistent Volume specification for a benchmark run
+type VolumeSpec struct {
+	// VolumeSource represents the source of the volume, e.g. an existing
+	// PVC, host path, git repo, etc.
+	VolumeSource corev1.VolumeSource `json:"volumeSource,omitempty"`
+
+	// PersistentVolumeClaim describes the persistent volume claim that will be
+	// created and used by the pod. This field *overrides* the VolumeSource to
+	// point to the created PVC
+	// +optional
+	PersistentVolumeClaim *PersistentVolumeClaimSpec `json:"persistentVolumeClaim,omitempty"`
```

Comment:
I think we should move PersistentVolumeClaim out of VolumeSpec, as it is a separate Kubernetes resource while VolumeSpec is connected to Pods.

Reply:
Since we need to validate them in relation to the other (volumeSpec.volumeSource.PVC.ClaimName == 'GENERATED' if volumeSpec.PVC is specified) it makes sense to group them together. This way we can attach the validator to the struct, making the validate easy.